controllers/openstackdataplane_controller.go

```diff
@@ -141,59 +130,58 @@ func (r *OpenStackDataPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 		logger.Info("Starting DataPlane deploy")
 		logger.Info("Set DeploymentReadyCondition false", "instance", instance)
 		instance.Status.Conditions.Set(condition.FalseCondition(condition.DeploymentReadyCondition, condition.RequestedReason, condition.SeverityInfo, condition.DeploymentReadyRunningMessage))
-		roles := &dataplanev1.OpenStackDataPlaneRoleList{}
+		nodeSets := &dataplanev1.OpenStackDataPlaneNodeSetList{}
 
-		listOpts := []client.ListOption{
-			client.InNamespace(instance.GetNamespace()),
-		}
-		labelSelector := map[string]string{
-			"openstackdataplane": instance.Name,
+		labelSelector := labels.NewSelector()
+		labelReq, err := labels.NewRequirement("openstackdataplane", selection.In, []string{instance.Name})
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to format labelSelector: %w", err)
 		}
-		if len(labelSelector) > 0 {
-			labels := client.MatchingLabels(labelSelector)
-			listOpts = append(listOpts, labels)
+		labelSelector.Add(*labelReq)
+
+		listOpts := client.ListOptions{
+			Namespace:     instance.GetNamespace(),
+			LabelSelector: labelSelector,
 		}
-		err = r.Client.List(ctx, roles, listOpts...)
+
+		err = r.Client.List(ctx, nodeSets, &listOpts)
 		if err != nil {
 			return ctrl.Result{}, err
 		}
-		logger.Info("found roles", "total", len(roles.Items))
-		if len(instance.Spec.Roles) > len(roles.Items) {
-			shouldRequeue = true
+
+		// If we didn't find any nodeSets with the labelSelector, we wont be able to deploy anything. Let's Log
+		// this and requeue.
+		if len(nodeSets.Items) == 0 {
+			logger.Info(fmt.Sprintf("No nodeSets were found with matching label: %v", labelSelector))
+			return ctrl.Result{RequeueAfter: time.Second * 5}, nil
 		}
 
-		for _, role := range roles.Items {
-			logger.Info("DataPlane deploy", "role.Name", role.Name)
-			if role.Spec.DataPlane != instance.Name {
-				err = fmt.Errorf("role %s: role.DataPlane does not match with role.Label", role.Name)
-				deployErrors = append(deployErrors, "role.Name: "+role.Name+" error: "+err.Error())
-			}
-			logger.Info("Role", "DeployStrategy.Deploy", role.Spec.DeployStrategy.Deploy, "Role.Namespace", instance.Namespace, "Role.Name", role.Name)
-			if !role.Spec.DeployStrategy.Deploy {
-				_, err := controllerutil.CreateOrPatch(ctx, helper.GetClient(), &role, func() error {
```

Comment:
I really think we need to put this back. Leave it out of scope for this change. No reason to change how NodeSets are deployed as part of this PR.